x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -452,6 +462,26 @@ export class Authenticator {
     this.options.featureUsageService.recordPreAccessAgreementUsage();
   }
 
+  getRequestOriginalURL(
+    request: KibanaRequest,
+    additionalQueryStringParameters?: Array<[string, string]>
+  ) {
+    const originalURLSearchParams = [
+      ...[...request.url.searchParams.entries()].filter(
+        ([key]) =>
+          key !== AUTH_URL_HASH_QUERY_STRING_PARAMETER &&
+          key !== AUTH_PROVIDER_HINT_QUERY_STRING_PARAMETER
```

Comment:
I'm really on the fence about this nit, but it _might_ be better if we abstract this out into a string array constant in "common/constants.ts" (`AUTH_METADATA_PARAMETERS`?) and change this bit to something like the following:
```suggestion
        ([key]) => !AUTH_METADATA_PARAMETERS.includes(key)
```

I don't know :) The SAML/OIDC auth providers are adding these query params, and this authenticator is stripping them out, with no strong coupling between them. I guess another alternative would be an interface or enum. See also my previous comment about the TSdoc comments which would help, I think.

Do with this what you will!

Reply:
Yeah, I'm torn on this too. Grouping these makes sense, but only in the context of `Authenticator`. I'm going to group them at this level for now, and we can re-consider this in the future if needed if it sounds good to you.